Fail to build on cygwin

Hi, I want to use cmcstl on my local cygwin but I can't build it.
Setup:
cygwin (32bit) 2.10.0
gcc 7.3
cmake 3.11.3
make 4.2.1
Running `make`  after `cmake .` failed in the test directory.
Output:

[  0%] Built target simple
[  0%] Built target common
[  1%] Built target range_access
[  2%] Built target optional
[  3%] Built target meta
[  4%] Built target headers
[  5%] Built target tagged
[  5%] Built target concepts.swap
[  6%] Built target concepts.core
[  7%] Built target concepts.fundamental
[  7%] Built target concepts.compare
[  8%] Built target concepts.object
[  9%] Built target concepts.range
[  9%] Built target concepts.iterator
[  9%] Built target raw_ptr
[  9%] Built target temporary_vector
[ 10%] Built target invoke
[ 10%] Built target not_fn
[ 11%] Built target counted_iterator
[ 12%] Built target common_iterator
[ 13%] Built target reverse_iterator
[ 14%] Built target iterator
[ 14%] Built target istreambuf_iterator
[ 15%] Built target move_iterator
[ 16%] Built target unreachable
[ 17%] Built target basic_iterator
[ 18%] Built target istream_iterator
[ 18%] Built target make_range
[ 18%] Built target iter.incomplete
[ 19%] Built target iter.operations
[ 19%] Built target ostream_iterator
[ 20%] Built target ostreambuf_iterator
[ 21%] Built target iter.any
[ 22%] Built target alg.upper_bound
[ 22%] Built target alg.unique_copy
[ 22%] Built target alg.transform
[ 22%] Built target alg.stable_partition
[ 23%] Built target alg.sort_heap
[ 23%] Built target alg.shuffle
[ 24%] Built target alg.swap_ranges
[ 25%] Built target alg.set_union3
[ 25%] Built target alg.set_union2
[ 26%] Built target alg.set_union1
[ 27%] Built target alg.set_symmetric_difference6
[ 28%] Built target alg.set_union4
[ 28%] Built target alg.set_symmetric_difference5
[ 28%] Built target alg.set_intersection6
[ 29%] Built target alg.set_intersection2
[ 30%] Built target alg.set_difference6
[ 31%] Built target alg.set_difference5
[ 31%] Built target alg.set_difference4
[ 32%] Built target alg.is_sorted_until
[ 32%] Built target alg.is_heap_until4
[ 33%] Built target alg.minmax_element
[ 34%] Built target alg.set_symmetric_difference2
[ 35%] Built target alg.is_heap_until3
[ 36%] Built target alg.prev_permutation
[ 36%] Built target alg.is_heap_until1
[ 37%] Built target alg.set_difference3
[ 37%] Built target alg.is_heap3
[ 38%] Built target alg.stable_sort
[ 38%] Built target alg.copy_backward
[ 39%] Built target alg.equal_range
[ 39%] Built target alg.inplace_merge
[ 40%] Built target alg.includes
[ 41%] Built target alg.nth_element
[ 41%] Built target alg.set_intersection1
[ 42%] Built target alg.is_partitioned
[ 43%] Built target alg.set_difference1
[ 43%] Built target alg.minmax
[ 44%] Built target alg.unique
[ 44%] Built target alg.find_if_not
[ 45%] Built target alg.set_intersection3
[ 46%] Built target alg.find_if
[ 47%] Built target alg.partition
[ 47%] Built target alg.generate_n
[ 48%] Built target alg.all_of
[ 48%] Built target alg.copy_n
[ 48%] Built target alg.set_intersection4
[ 48%] Built target alg.any_of
[ 49%] Built target alg.generate
[ 50%] Built target alg.copy
[ 51%] Built target alg.binary_search
[ 51%] Built target alg.set_union5
[ 52%] Built target alg.is_heap4
[ 52%] Built target alg.reverse
[ 52%] Built target alg.find_end
[ 53%] Built target alg.set_symmetric_difference1
[ 54%] Built target alg.copy_if
[ 55%] Built target alg.sort
[ 56%] Built target alg.set_union6
[ 56%] Built target alg.remove_if
[ 57%] Built target alg.count_if
[ 57%] Built target alg.is_sorted
[ 58%] Built target alg.find_first_of
[ 58%] Built target alg.adjacent_find
[ 58%] Built target alg.rotate
[ 59%] Built target alg.fill
[ 60%] Built target alg.merge
[ 60%] Built target alg.partial_sort
[ 61%] Built target alg.replace_if
[ 62%] Built target alg.is_heap_until2
[ 62%] Built target alg.min
[ 63%] Built target alg.is_heap1
[ 64%] Built target alg.replace
[ 65%] Built target alg.is_permutation
[ 66%] Built target alg.sample
[ 66%] Built target alg.equal
[ 66%] Built target alg.fill_n
[ 67%] Built target alg.find
[ 67%] Built target alg.set_difference2
[ 68%] Built target alg.is_heap2
[ 69%] Built target alg.lower_bound
[ 69%] Built target alg.max
[ 70%] Built target alg.search_n
[ 71%] Built target alg.max_element
[ 72%] Built target alg.for_each
[ 73%] Built target alg.partial_sort_copy
[ 74%] Built target alg.min_element
[ 75%] Built target alg.set_intersection5
[ 76%] Built target alg.make_heap
[ 76%] Built target alg.remove
[ 76%] Built target alg.search
[ 77%] Built target alg.mismatch
[ 78%] Built target alg.none_of
[ 78%] Built target alg.move
[ 79%] Built target alg.move_backward
[ 79%] Built target alg.partition_copy
[ 79%] Built target alg.lexicographical_compare
[ 80%] Built target alg.partition_point
[ 81%] Built target alg.set_symmetric_difference4
[ 81%] Built target alg.pop_heap
[ 81%] Built target alg.set_symmetric_difference3
[ 82%] Built target alg.push_heap
[ 82%] Built target alg.next_permutation
[ 83%] Built target alg.count
[ 84%] Built target alg.remove_copy
[ 85%] Built target alg.remove_copy_if
[ 85%] Built target alg.replace_copy
[ 86%] Built target alg.replace_copy_if
[ 87%] Built target alg.reverse_copy
[ 88%] Built target alg.rotate_copy
[ 89%] Built target view.filter
[ 89%] Built target view.counted
[ 89%] Built target view.repeat_n
[ 89%] Building CXX object test/view/CMakeFiles/view.bounded.dir/bounded_view.cpp.o
In file included from /home/sharon1/external_libs/cmcstl2/test/view/bounded_view.cpp:12:0:
/home/sharon1/external_libs/cmcstl2/include/stl2/view/bounded.hpp:95:31: error: ‘fn’ in namespace ‘::’ does not name a type
   inline constexpr __bounded::fn bounded {};
                               ^~
/home/sharon1/external_libs/cmcstl2/include/stl2/view/bounded.hpp:95:44: error: extra ‘;’ [-Werror=pedantic]
   inline constexpr __bounded::fn bounded {};
                                            ^
/home/sharon1/external_libs/cmcstl2/test/view/bounded_view.cpp: In function ‘int main()’:
/home/sharon1/external_libs/cmcstl2/test/view/bounded_view.cpp:23:23: error: ‘bounded’ is not a member of ‘std::experimental::ranges::v1::view’
   auto x = rg | view::bounded;
                       ^~~~~~~
/home/sharon1/external_libs/cmcstl2/test/view/bounded_view.cpp:23:23: note: suggested alternative: ‘counted’
   auto x = rg | view::bounded;
                       ^~~~~~~
                       counted
/home/sharon1/external_libs/cmcstl2/test/view/bounded_view.cpp:25:25: error: template argument 1 is invalid
   static_assert(models::View<decltype(x)>);
                         ^~~~~~~~~~~~~~~~~
/home/sharon1/external_libs/cmcstl2/test/view/bounded_view.cpp:26:25: error: template argument 1 is invalid
   static_assert(models::SizedRange<decltype(x)>);
                         ^~~~~~~~~~~~~~~~~~~~~~~
/home/sharon1/external_libs/cmcstl2/test/view/bounded_view.cpp:27:25: error: template argument 1 is invalid
   static_assert(models::BoundedRange<decltype(x)>);
                         ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/sharon1/external_libs/cmcstl2/test/view/bounded_view.cpp:28:25: error: template argument 1 is invalid
   static_assert(models::RandomAccessRange<decltype(x)>);
                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sharon1/external_libs/cmcstl2/test/view/bounded_view.cpp:32:65: error: ‘bounded’ is not a member of ‘std::experimental::ranges::v1::view’
   auto x = view::counted(bidirectional_iterator(rg), 5) | view::bounded;
                                                                 ^~~~~~~
/home/sharon1/external_libs/cmcstl2/test/view/bounded_view.cpp:32:65: note: suggested alternative: ‘counted’
   auto x = view::counted(bidirectional_iterator(rg), 5) | view::bounded;
                                                                 ^~~~~~~
                                                                 counted
/home/sharon1/external_libs/cmcstl2/test/view/bounded_view.cpp:34:25: error: template argument 1 is invalid
   static_assert(models::View<decltype(x)>);
                         ^~~~~~~~~~~~~~~~~
/home/sharon1/external_libs/cmcstl2/test/view/bounded_view.cpp:35:25: error: template argument 1 is invalid
   static_assert(models::SizedRange<decltype(x)>);
                         ^~~~~~~~~~~~~~~~~~~~~~~
/home/sharon1/external_libs/cmcstl2/test/view/bounded_view.cpp:36:25: error: template argument 1 is invalid
   static_assert(models::BoundedRange<decltype(x)>);
                         ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/sharon1/external_libs/cmcstl2/test/view/bounded_view.cpp:37:25: error: template argument 1 is invalid
   static_assert(models::ForwardRange<decltype(x)>);
                         ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/sharon1/external_libs/cmcstl2/test/view/bounded_view.cpp:38:26: error: template argument 1 is invalid
   static_assert(!models::BidirectionalRange<decltype(x)>);
                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sharon1/external_libs/cmcstl2/test/view/bounded_view.cpp:39:58: error: ‘bounded’ is not a member of ‘std::experimental::ranges::v1::view’
   static_assert(models::Same<decltype(x), decltype(view::bounded(x))>);
                                                          ^~~~~~~
/home/sharon1/external_libs/cmcstl2/test/view/bounded_view.cpp:39:58: note: suggested alternative: ‘counted’
   static_assert(models::Same<decltype(x), decltype(view::bounded(x))>);
                                                          ^~~~~~~
                                                          counted
/home/sharon1/external_libs/cmcstl2/test/view/bounded_view.cpp:39:58: error: ‘bounded’ is not a member of ‘std::experimental::ranges::v1::view’
/home/sharon1/external_libs/cmcstl2/test/view/bounded_view.cpp:39:58: note: suggested alternative: ‘counted’
   static_assert(models::Same<decltype(x), decltype(view::bounded(x))>);
                                                          ^~~~~~~
                                                          counted
/home/sharon1/external_libs/cmcstl2/test/view/bounded_view.cpp:39:25: error: template argument 1 is invalid
   static_assert(models::Same<decltype(x), decltype(view::bounded(x))>);
                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sharon1/external_libs/cmcstl2/test/view/bounded_view.cpp:39:25: error: template argument 2 is invalid
cc1plus: all warnings being treated as errors
make[2]: *** [test/view/CMakeFiles/view.bounded.dir/build.make:63: test/view/CMakeFiles/view.bounded.dir/bounded_view.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:6787: test/view/CMakeFiles/view.bounded.dir/all] Error 2
make: *** [Makefile:141: all] Error 2

--- a/include/stl2/view/bounded.hpp
+++ b/include/stl2/view/bounded.hpp
@@ -79,7 +79,7 @@ STL2_OPEN_NAMESPACE {
 	} // namespace ext
 
 	namespace view {
-		namespace __bounded {
+		namespace __stl2_bounded {
 			struct fn : detail::__pipeable<fn> {
 				template <ext::ViewableRange R>
 				requires BoundedRange<R>
@@ -90,9 +90,9 @@ STL2_OPEN_NAMESPACE {
 				constexpr auto operator()(R&& r) const
 				{ return ext::bounded_view{std::forward<R>(r)}; }
 			};
-		} // namespace __bounded
+		} // namespace __stl2_bounded
 
-		inline constexpr __bounded::fn bounded {};
+		inline constexpr __stl2_bounded::fn bounded {};
 	} // namespace view
 } STL2_CLOSE_NAMESPACE
 

